src/drivers/distance_sensor/hc_sr04/hc_sr04.cpp

```diff
@@ -85,138 +98,105 @@ class HC_SR04 : public cdev::CDev
 	HC_SR04();
 	virtual ~HC_SR04();
 
-	virtual int 		init();
+	virtual int init();
+
+	void interrupt(unsigned time);
 
-	virtual ssize_t		read(device::file_t *filp, char *buffer, size_t buflen);
-	virtual int			ioctl(device::file_t *filp, int cmd, unsigned long arg);
+	virtual int ioctl(device::file_t *filp, int cmd, unsigned long arg) override;
```

Comment:
Things like the ioctl, read, and ringbuffers can usually be gutted entirely. These were used historically to read directly from the device (eg `::open("/dev/distance_sensor0")`), but all of that has moved entirely to orb.